controllers/drcluster_controller.go

```diff
@@ -40,7 +40,8 @@ import (
 
 // DRClusterReconciler reconciles a DRCluster object
 type DRClusterReconciler struct {
-	client.Client
+	client.Writer
+	client.StatusWriter
```

Comment:
The create,update,delete, etc are already a client.Writer members.  In other words, there is no need to explicitly call the Update() function with client.Writer object.  client.Client.Update() will do the same the same.